docs/design/configs.md

```diff
@@ -0,0 +1,130 @@
+# Generic Config Object
+
+## Abstract
+
+A generic ``Config`` object to store command-line flags and environment variables
+that is used for configuration of ``Kubernetes`` components living on top of ``etcd``
+is proposed in this proposal.
```

Comment:
isn't the current PR (map[string]string bascially exactly like secrets, except that secrets uses []byte as the value because binary.  Do we want to support binary params?